cumulus/client/cirrus-executor/src/lib.rs

```diff
@@ -353,32 +402,8 @@ where
 		};
 
 		// A bundle equivocation occurs.
-		if let Some(bundle_equivocation_proof) = check_equivocation(bundle) {
-			let mut overseer_handle = self.overseer_handle.clone();
-			self.spawner.spawn(
-				"cirrus-submit-bundle-equivocation-proof",
-				None,
-				async move {
-					tracing::debug!(
-						target: LOG_TARGET,
-						"Submitting bundle equivocation proof in a background task..."
-					);
-					overseer_handle
-						.send_msg(
-							CollationGenerationMessage::BundleEquivocationProof(
-								bundle_equivocation_proof,
-							),
-							"SubmitBundleEquivocationProof",
-						)
-						.await;
-					tracing::debug!(
-						target: LOG_TARGET,
-						"Bundle equivocation proof submission finished"
-					);
-				}
-				.boxed(),
-			);
-
+		if let Some(equivocation_proof) = check_equivocation(bundle) {
+			self.submit_bundle_equivocation_proof(equivocation_proof);
```

Comment:
Why moving implementation out of here?
It is not reused anywhere and doesn't help with readability because not you have to jump over 200 lines up for no reason as far as I'm concerned.

Reply:
Because it's easier for me to maintain, I prefer the _core_ function which tells the entire data flow can just fit into my screen so that I don't have to scroll to have an overview, providing a relatively dedicated thinking context.

Being not reused does not mean it doesn't help the readability:

1. When you extract a chunk of code to a small function, you have to think about the function name. With a proper function name, which implicitly functions as _an inline comment_, it does help understand the code.
2. Jumping is zero cost for me. So again, it really depends on the developer.

Reply:
Those functions have just "self.spawner.spawn" with a single call to overseer in them + some helpful logging. Jumping might be low cost, but never zero. I'm fine merging as is, but only writing code that is convenient to you is not very sustainable long term.

Reply:
Once we have more people jump in, I totally agree that it's valuable to collectively set up some common rules in the group that is better for maintainability and sustainability in the long run. Otherwise, it's just personal preference, it's not math, only right or wrong.